lib/kafo/puppet_module.rb

```diff
@@ -12,7 +12,11 @@ class PuppetModule
     attr_reader :name, :identifier, :params, :dir_name, :class_name, :manifest_name, :manifest_path,
                 :groups, :params_path, :params_class_name, :configuration, :raw_data
 
-    def initialize(identifier, parser = KafoParsers::PuppetModuleParser, configuration = KafoConfigure.config)
+    def self.find_parser
+      @@parser ||= KafoParsers::Parsers.find_available(:logger => KafoConfigure.logger) || :none
```

Comment:
why class variable? I don't see it used anywhere else, for memoization instance variable should be fine